### ISSUE TITLE
Regarding the reproduction of RTE results from Table 2

### ISSUE THREAD

**Issue by logicraft206** [MIkumikumi0116/DoRA#4]
Hello, thank you for your outstanding work on DoRA and for making the code publicly available.

I am currently trying to reproduce the results for the RTE and CoLA tasks as reported in Table 2.

Following the details in your paper, I used the hyperparameters specified in Table 8 as the sole configuration for my experiments. I ran the experiments using the 5 different random seeds you have set in the code to calculate the average score.

Here are the averaged results I obtained:

    RTE: 79.42

    CoLA: 64.67

The result for CoLA (64.67) appears to be quite close to what's reported in the paper, which is great. However, my RTE result (79.42) seems to have a noticeable gap compared to the score in Table 2.

I have double-checked my setup, but I am wondering if I might have missed any crucial details or if there are other specific configurations for the RTE task that I should be aware of.

Could you please provide some guidance on this? Any suggestions or insights you could offer would be greatly appreciated.

Thank you!

**Comment by MIkumikumi0116** [comment]
Please post the training loss and evaluation metric curves. They are located in Log/{{task_tag}} (defined by the hyperparameter values). If the loss curve fails to decrease, there might be some hidden issues.

Is there a problem with the results on other datasets? RTE is the smallest dataset and is most affected by randomness in the training process. If you experiment with different random seeds on multiple datasets, you should observe that the results on MNLI, the largest dataset, are very consistent across different seeds, while RTE shows larger fluctuations.

PyTorch's random seeds only guarantee consistency under the same environment—including GPU hardware, operating system, Python and PyTorch versions, etc [link](https://docs.pytorch.org/docs/stable/notes/randomness.html). If your experimental configuration differs from mine, I suspect the impact would be comparable to changing the random seed. This might affect the training results on RTE, though a 2% degradation does seem excessively large based on experience.

I suggest you run several more experiments with different random seeds on RTE and observe the fluctuations in training results. If the fluctuations are large (e.g., 2%, 3%, or higher), the metric degradation you observed might just be normal variation.

Additionally, as mentioned in [issue2](https://github.com/MIkumikumi0116/DoRA/issues/2), the code for downloading GLUE in the repo is no longer functional. Have you fixed this issue? If so, could you please submit a PR? This would help other researchers, and I would greatly appreciate it.

**Comment by logicraft206** [comment]
Hi@MIkumikumi0116 , thank you very much for your detailed and swift reply!

As you requested, I've attached the training and evaluation metric curves from one of the RTE runs below. Looking at the evaluation curves, I agree they show some instability.

<img width="4000" height="800" alt="Image" src="https://github.com/user-attachments/assets/e32ccce6-0f4b-429f-b657-823d56261e55" />

Following your advice, I also ran another set of experiments with 5 different, consecutive random seeds. The new average accuracy I obtained for RTE is 80.43.

Although this result is higher, there is still a noticeable gap of about 1.3% compared to the paper's. Given this and the unstable evaluation curves, do you have any other suggestions or insights you might recommend?

Regarding the data preprocessing and issue2: I have not fixed the issue with the download script. My workaround was to download the dataset directly from the official GLUE benchmark website, convert it into the Hugging Face datasets format, and then apply the preprocessing logic from your repository. Since my CoLA result was quite close to the paper's, I assumed this data pipeline was acceptable.

Thank you again for your time and guidance!
